contracts/managers/SocialTradingManager.sol

```diff
@@ -200,9 +220,63 @@ contract SocialTradingManager is
             _newAllocation
         );
 
+        // Save new allocation
         pools[address(_tradingPool)].currentAllocation = _newAllocation;
     }
 
+    /*
+     * Start fee update process, set fee update timestamp and commit to new fee.
+     *
+     * @param _tradingPool        The address of the trading pool being updated
+     * @param _newEntryFee        New entry fee in a scaled decimal value
+     *                              (e.g. 100% = 1e18, 1% = 1e16)
+     */
+    function initiateSetEntryFee(
+        IRebalancingSetTokenV2 _tradingPool,
+        uint256 _newEntryFee
+    )
+        external
+        onlyTrader(_tradingPool)
+    {
+        // Validate new entry fee doesn't exceed max
+        validateNewEntryFee(_newEntryFee);
+
+        // Log new entryFee and timestamp to start timelock from
+        pools[address(_tradingPool)].feeUpdateTimestamp = block.timestamp.add(feeUpdateTimelock);
+        pools[address(_tradingPool)].newEntryFee = _newEntryFee;
+    }
+
+    /*
+     * Finalize fee update, change fee on RebalancingSetTokenV2 if time lock period has elapsed.
+     *
+     * @param _tradingPool        The address of the trading pool being updated
+     */
+    function finalizeSetEntryFee(
+        IRebalancingSetTokenV2 _tradingPool
+    )
+        external
+        onlyTrader(_tradingPool)
+    {
+        // If feeUpdateTimestamp is equal to 0 indicates initiate wasn't called
+        require(
+            feeUpdateTimestamp(_tradingPool) != 0,
+            "SocialTradingManager.finalizeSetFeeRecipient: Must initiate fee change first."
+        );
+
+        // Current block timestamp must exceed feeUpdateTimestamp
+        require(
+            block.timestamp >= feeUpdateTimestamp(_tradingPool),
+            "SocialTradingManager.finalizeSetFeeRecipient: Time lock period must elapse to update fees."
+        );
+
+        // Save entryFee to memory and reset fee related attributes
+        uint256 newEntryFee = newEntryFee(_tradingPool);
+        pools[address(_tradingPool)].feeUpdateTimestamp = 0;
+        pools[address(_tradingPool)].newEntryFee = 0;
+
+        _tradingPool.setEntryFee(newEntryFee); 
```

Comment:
Why not just do the following in one line: `_tradingPool.setEntryFee(newEntryFee(_tradingPool)`?

Reply:
I guess that one should be fine, updating the timestamp should be enough to stop any reentrancy type attack